api/provider/aws/test_aws/environment_delete_test.go

```diff
@@ -117,3 +119,50 @@ func TestDeleteEnvironmentIdempotence(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestCheckEnvironmentDependencies(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+
+	mockAWS := awsc.NewMockClient(ctrl)
+	tagStore := tag.NewMemoryStore()
+	mockConfig := mock_config.NewMockAPIConfig(ctrl)
+
+	mockConfig.EXPECT().Instance().Return("test").AnyTimes()
+
+	envTag := models.Tag{
+		EntityID:   "env_id",
+		EntityType: "environment",
+		Key:        "name",
+		Value:      "env_name",
+	}
+
+	if err := tagStore.Insert(envTag); err != nil {
+		t.Fatal(err)
+	}
+
+	cases := []string{"service", "task", "load_balancer"}
+	for _, c := range cases {
+		tag := models.Tag{
+			EntityID:   "id",
+			EntityType: c,
+			Key:        "environment_id",
+			Value:      "env_id",
+		}
+
+		if err := tagStore.Insert(tag); err != nil {
+			t.Fatal(err)
+		}
+
+		target := provider.NewEnvironmentProvider(mockAWS.Client(), tagStore, mockConfig)
+		if err := target.Delete("env_id"); err != nil {
+			if !strings.Contains(err.Error(), errors.DependencyError.String()) {
+				t.Errorf("Error: %s", err)
+			}
+		}
+
+		if err := tagStore.Delete(tag.EntityType, tag.EntityID, tag.Key); err != nil {
```

Comment:
I think it would be cleaner to create a new `tagStore` for each case instead of doing cleanup. This may be more of a personal philosophy, but I think readability is more important than speed/efficiency for unit tests.

Reply:
Yeah I was on the fence about this but I agree.